test/allocators/AllocatorWithAirnode.sol.js

```diff
@@ -1,36 +1,30 @@
-const hre = require('hardhat');
+const { ethers } = require('hardhat');
+const helpers = require('@nomicfoundation/hardhat-network-helpers');
 const { expect } = require('chai');
 const testUtils = require('../test-utils');
 
 describe('AllocatorWithAirnode', function () {
-  let roles;
-  let accessControlRegistry, allocatorWithAirnode;
-  let allocatorWithAirnodeAdminRoleDescription = 'AllocatorWithAirnode admin role';
-  let slotSetterRoleDescription = 'Slot setter';
-  let airnodeSlotSetterRole;
-  let slotIndex = Math.floor(Math.random() * 1000);
-  let subscriptionId, anotherSubscriptionId;
-  let expirationTimestamp;
-
-  beforeEach(async () => {
-    const accounts = await hre.ethers.getSigners();
-    roles = {
+  async function deploy() {
+    const accounts = await ethers.getSigners();
+    const roles = {
       deployer: accounts[0],
       airnode: accounts[1],
       slotSetter: accounts[2],
       anotherSlotSetter: accounts[3],
       randomPerson: accounts[9],
     };
-    const accessControlRegistryFactory = await hre.ethers.getContractFactory('AccessControlRegistry', roles.deployer);
-    accessControlRegistry = await accessControlRegistryFactory.deploy();
-    const allocatorWithAirnodeFactory = await hre.ethers.getContractFactory('AllocatorWithAirnode', roles.deployer);
-    allocatorWithAirnode = await allocatorWithAirnodeFactory.deploy(
+    const allocatorWithAirnodeAdminRoleDescription = 'AllocatorWithAirnode admin';
+    const slotSetterRoleDescription = 'Slot setter';
+    const accessControlRegistryFactory = await ethers.getContractFactory('AccessControlRegistry', roles.deployer);
+    const accessControlRegistry = await accessControlRegistryFactory.deploy();
+    const allocatorWithAirnodeFactory = await ethers.getContractFactory('AllocatorWithAirnode', roles.deployer);
+    const allocatorWithAirnode = await allocatorWithAirnodeFactory.deploy(
       accessControlRegistry.address,
       allocatorWithAirnodeAdminRoleDescription
     );
     const airnodeRootRole = testUtils.deriveRootRole(roles.airnode.address);
-    const airnodeAdminRole = await allocatorWithAirnode.deriveAdminRole(roles.airnode.address);
-    airnodeSlotSetterRole = await allocatorWithAirnode.deriveSlotSetterRole(roles.airnode.address);
+    const airnodeAdminRole = testUtils.deriveRole(airnodeRootRole, allocatorWithAirnodeAdminRoleDescription);
```

Comment:
(Kind of related to this change) Can you please remind me why AllocatorWithAirnode.sol does not call `_deriveAdminRole()` in [AccessControlRegistryAdminned.sol](https://github.com/api3dao/airnode-protocol-v1/blob/refactor-allocators/contracts/access-control-registry/AccessControlRegistryAdminned.sol#L51) like [RequesterAuthorizerWithAirnode.sol](https://github.com/api3dao/airnode-protocol-v1/blob/refactor-allocators/contracts/authorizers/RequesterAuthorizerWithAirnode.sol#L111) does?

Reply:
You're right, it should. Related to this, I'm planning to remove all public functions that expose roles (such as `deriveAdmin()` of AllocatorWithAirnode) because they can be derived off-chain/in the contract that need them. No one will make an external call if they can avoid it, I was being silly by cluttering the interface with these. Migrating these to `testUtils` was in fact preparation for that.

Reply:
I forgot to address this so created https://github.com/api3dao/airnode-protocol-v1/issues/86